S7.Net/Types/Timer.cs

```diff
@@ -13,26 +13,30 @@ public static class Timer
         public static double FromByteArray(byte[] bytes)
         {
             double wert = 0;
-            Int16 value = (Int16)Types.Word.FromBytes(bytes[1], bytes[0]);
-            string txt = Conversion.ValToBinString(value);
-            wert = Conversion.BinStringToInt32(txt.Substring(4, 4)) * 100.0;
-            wert += Conversion.BinStringToInt32(txt.Substring(8, 4)) * 10.0;
-            wert += Conversion.BinStringToInt32(txt.Substring(12, 4));
-            switch (txt.Substring(2, 2))
+
+            wert = ((bytes[0]) & 0x0F) * 100.0;
+            wert += ((bytes[1] >> 4) & 0x0F) * 10.0;
+            wert += ((bytes[1]) & 0x0F) * 1.0;
+
+            /// this value is not used... may for a nother exponation
+            ///int unknown = (bytes[0] >> 6) & 0x03;
+
+            switch ((bytes[0] >> 4) & 0x03)
             {
-                case "00":
+                case 0:
                     wert *= 0.01;
                     break;
-                case "01":
+                case 1:
```

Comment:
Using binary literals might be more obvious 

```csharp
case 0b00:
    break;
case 0b01:
    break;
case 0b10:
    break;
case 0b11:
    break;
```

Reply:
Bit values are good for displaying FLAGs where every bit has a different meaning but in this case it is more or less a ENUM or even a exponentiation: ( wert *= 10^(n-2) ) so Int values are good I think?!